awscli/customizations/s3/utils.py

```diff
@@ -493,22 +493,23 @@ def _decode_keys(self, parsed, **kwargs):
 class ScopedEventHandler(object):
     """Register an event callback for the duration of a scope."""
 
-    def __init__(self, session, event_name, handler, unique_id=None,
+    def __init__(self, event_emitter, event_name, handler, unique_id=None,
                  unique_id_uses_count=False):
-        self._session = session
+        self._event_emitter = event_emitter
         self._event_name = event_name
         self._handler = handler
         self._unique_id = unique_id
         self._unique_id_uses_count = unique_id_uses_count
 
     def __enter__(self):
-        self._session.register(self._event_name, self._handler, self._unique_id,
-                               self._unique_id_uses_count)
+        self._event_emitter.register(self._event_name, self._handler,
+                                     self._unique_id,
+                                     self._unique_id_uses_count)
```

Comment:
I thought by switching over to clients that I could remove ``unique_id_uses_count`` argument. I originally did the change but I saw test failures on the test that was added to catch that a handler is not deregistered too early when two ScopedEventHandlers are being used to listing objects. By switching to clients I thought that event emitters across clients were separate, so registering and deregistering on one client did not affect the other. We may need to look into it.